File: DESAFIO.BLOG.Domain/Entities/ChatMessage.cs
using DESAFIO.BLOG.Domain.Entities;

namespace DESAFIO.BLOG.Domain.Entities
{
    public class ChatMessage
    {
        public Guid Id { get; set; }
        public string Content { get; set; }
        public DateTime SentAt { get; set; }
        public Guid SenderId { get; set; }
        public ApplicationUser Sender { get; set; }
        public Guid ReceiverId { get; set; }
        public Guid ChatId { get; set; }
        public Chat Chat { get; set; }
        public ApplicationUser Receiver { get; set; }
    }
}

File: Front.Angular/src/app/Models/ChatMessage.ts
import { ApplicationUser } from "./ApplicationUser";
import { Chat } from "./Chat";

export interface ChatMessage {
  id: string;
  content: string;
  sentAt: Date;
  senderId: string;
  sender: ApplicationUser;
  receiverId: string;
  chatId: string;
  chat: Chat;
  receiver: ApplicationUser;
}
